sql/2020/22_HTTP_2/avg_number_of_h2_pushed_resources_and_avg_bytes.sql

```diff
@@ -0,0 +1,26 @@
+#standardSQL
+# 22_HTTP_2 - Average number of HTTP/2 Pushed Resources and Average Bytes
+SELECT 
+  client,
+  COUNT(DISTINCT page) AS num_pages,
+  ROUND(AVG(num_requests),2) AS avg_pushed_requests,
```

Comment:
Would percentiles be more useful? Averages tend to be easily skewed. For example, the median calculation would look like this:
```suggestion
  APPROX_QUANTILES(num_requests, 1000)[OFFSET(500)] AS median_pushed_requests,
```

(The file would also need to be renamed accordingly)

Many chapters also generate a few significant percentiles at the same time using a pattern like this:

```sql
SELECT
  percentile,
  client,
  APPROX_QUANTILES(metric, 1000)[OFFSET(percentile * 10)] AS value
FROM
  #(table or query),
  UNNEST([10, 25, 50, 75, 90]) AS percentile
GROUP BY
  percentile,
  client
ORDER BY
  percentile,
  client
```

Reply:
I agree that averages are not the best. Following your suggestions and guidance I have done the following:

1. Renamed `avg_number_of_h2_pushed_resources_and_avg_bytes_by_content_type` to `number_of_h2_pushed_resources_and_avg_bytes_by_content_type`

2.